includes/class-wc-gateway-ppec-cart-handler.php

```diff
@@ -38,6 +43,83 @@ public function before_cart_totals() {
 	}
 
 	/**
+	 * Generates the cart for express checkout on a product level.
+	 *
+	 * @since 1.4.0
+	 */
+	public function generate_cart() {
+		if ( ! wp_verify_nonce( $_POST['nonce'], '_wc_ppec_generate_cart_nonce' ) ) {
+			wp_die( __( 'Cheatin&#8217; huh?', 'woocommerce-gateway-paypal-express-checkout' ) );
+		}
+
+		if ( ! defined( 'WOOCOMMERCE_CART' ) ) {
+			define( 'WOOCOMMERCE_CART', true );
+		}
+
+		WC()->shipping->reset_shipping();
+
+		global $post;
+
+		$product = wc_get_product( $post->ID );
+		$qty     = ! isset( $_POST['qty'] ) ? 1 : absint( $_POST['qty'] );
+
+		/**
+		 * If this page is single product page, we need to simulate
+		 * adding the product to the cart taken account if it is a
+		 * simple or variable product.
+		 */
+		if ( is_single() ) {
+			// First empty the cart to prevent wrong calculation.
+			WC()->cart->empty_cart();
```

Comment:
can we make it works without emptying the cart? E.g. a customer has put A and B in the cart then goes to C and then clicking the button from there?